### PR TITLE
Athena Neptune Connector - support to connect IAM enabled Neptune cluster

### PR DESCRIPTION
**Pull request by abhishekpradeepmishra**
*Issue #, if available:*

*Description of changes:*

Added support to connect IAM enabled Neptune cluster

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: athena-neptune/athena-neptune.yaml [hunk 1]
```diff
@@ -72,6 +75,8 @@ Resources:
           neptune_endpoint: !Ref NeptuneClusterEndpoint
           neptune_port: !Ref NeptunePort
           neptune_cluster_res_id: !Ref NeptuneClusterResourceID
+          iam_enabled: !Ref IAMEnabled
+          SERVICE_REGION: !Ref AWS::Region
```
Comment (avirtuos) [inline athena-neptune/athena-neptune.yaml:79]: Why is this needed? Can't this be derived from the Lambda runtime env?

File: athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/NeptuneConnection.java [hunk 2]
```diff
@@ -21,21 +21,27 @@
...
 import org.apache.tinkerpop.gremlin.process.traversal.AnonymousTraversalSource;
 import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversalSource;
 
 public class NeptuneConnection
 {
-     private static Cluster cluster = null;
+    private static Cluster cluster = null;
 
-    NeptuneConnection(String neptuneEndpoint, String neptunePort)
+    NeptuneConnection(String neptuneEndpoint, String neptunePort, Boolean enabledIAM)
```
Comment (avirtuos) [inline athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/NeptuneConnection.java:33]: Why Boolean (boxed type) as opposed to Boolean (non-boxed)?

Comment (avirtuos) [reply athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/NeptuneConnection.java:33]: You may need to add a null check somewhere in order to use the non-boxed form but I think that is preferred for how you are using this as null seems to be akin to false for this usage.

**Review by avirtuos — COMMENTED** [review]
Two minor comments but this seems ready to ship otherwise.

File: athena-neptune/athena-neptune.yaml (hunk 1 above)
Comment (abhishekpradeepmishra) [reply athena-neptune/athena-neptune.yaml:79]: SERVICE_REGION is needed by the SignV4 SDK SDK

https://docs.aws.amazon.com/neptune/latest/userguide/iam-auth-connecting-gremlin-java.html
https://github.com/aws/amazon-neptune-gremlin-java-sigv4/blob/master/src/main/java/com/amazon/neptune/gremlin/driver/sigv4/ChainedSigV4PropertiesProvider.java

File: athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/NeptuneConnection.java (hunk 2 above)
Comment (abhishekpradeepmishra) [reply athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/NeptuneConnection.java:33]: can convert this to primitive type boolean.  yes, the default value is false, even if user doesn't supply a value. Boolean.parseBoolean(System.getenv("iam_enabled")) will return false.

**Review by avirtuos — APPROVED** [review]
